src/payment/coupon/data/sagas.js

```diff
@@ -23,12 +23,16 @@ export function* handleAddCoupon(action) {
   try {
     const result = yield call(postCoupon, action.payload.code);
     yield put(fetchBasketSuccess(result));
-    if (result.voucher === undefined) {
+    if (result.coupons === undefined) {
```

Comment:
Do we know that this array will be undefined still?  Or will it be an empty array?

Reply:
Empty array, changed to `result.coupons.length === 0`